app/scenes/Settings/components/SharesTable.tsx

```diff
@@ -1,108 +1,137 @@
-import { observer } from "mobx-react";
+import compact from "lodash/compact";
 import * as React from "react";
 import { useTranslation } from "react-i18next";
 import { unicodeCLDRtoBCP47 } from "@shared/utils/date";
 import Share from "~/models/Share";
 import { Avatar } from "~/components/Avatar";
 import Flex from "~/components/Flex";
-import TableFromParams from "~/components/TableFromParams";
+import {
+  type Props as TableProps,
+  SortableTable,
+} from "~/components/SortableTable";
+import { type Column as TableColumn } from "~/components/Table";
 import Time from "~/components/Time";
 import useUserLocale from "~/hooks/useUserLocale";
 import ShareMenu from "~/menus/ShareMenu";
 import { formatNumber } from "~/utils/language";
 
-type Props = Omit<React.ComponentProps<typeof TableFromParams>, "columns"> & {
-  data: Share[];
+type Props = Omit<
+  TableProps<Share>,
+  "columns" | "rowHeight" | "gridColumns"
+> & {
   canManage: boolean;
 };
 
-function SharesTable({ canManage, data, ...rest }: Props) {
+export function SharesTable({ data, canManage, ...rest }: Props) {
   const { t } = useTranslation();
   const language = useUserLocale();
   const hasDomain = data.some((share) => share.domain);
 
-  const columns = React.useMemo(
+  const gridColumns = React.useMemo(() => {
+    if (canManage && hasDomain) {
+      return "4fr 2fr 2fr 2fr 1.5fr 1fr 0.5fr"; // all columns will be displayed.
```

Comment:
Same as `PeopleTable`.. I'm not completely sure how to improve this.
One iteration of the implementation had this width config as part of the individual column defs itself - but I figured having the grid styles together (like how it's in CSS) provided better understanding.

Reply:
As above, definitely prefer figuring out how to get the width as a config